File: examples/napi-test-simpleAgents/handlers.ts
const stakeholderMap: Record<string, string> = {
  google: "Sundar Pichai",
  microsoft: "Satya Nadella",
  apple: "Tim Cook",
  amazon: "Andy Jassy",
};

export type CustomWorkerRequest = {
  handler: string;
  handlerFile?: string;
  payload: unknown;
  context: unknown;
};

function getSellerNameFromPayload(payload: unknown): string {
  if (payload === null || typeof payload !== "object") {
    return "unknown";
  }
  const raw = (payload as Record<string, unknown>).company_name;
  if (raw === undefined || raw === null) {
    return "unknown";
  }
  const key = String(raw).trim().toLowerCase();
  if (!key) {
    return "unknown";
  }
  return stakeholderMap[key] ?? "unknown";
}

/** Pass this as `opts.customWorker` / `customWorkerDispatch`. */
export function customWorkerDispatch(req: CustomWorkerRequest): string {
  if (req.handler === "get_seller_name") {
    return getSellerNameFromPayload(req.payload);
  }
  throw new Error(`unknown custom_worker handler: ${req.handler}`);
}

File: examples/python-test-simpleAgents/evals/invoice/handlers.py
# Canonical copy lives at `../../workflows/email-classification/handlers.py`.
# Duplicated because `run_eval_suite` resolves `handlers.py` relative to the eval suite YAML directory.


def get_seller_name(context, payload):
    """`context`: workflow context from the runner; `payload`: this node's `config.payload` (see test.yaml)."""
    # print(f"context: {context}")
    # print(f"payload: {payload}")
    company_name = None
    if isinstance(payload, dict):
        company_name = payload.get("company_name")
    company_name = str(company_name or "").strip().lower()

    stakeholder_map = {
        "google": "Sundar Pichai",
        "microsoft": "Satya Nadella",
        "apple": "Tim Cook",
        "amazon": "Andy Jassy",
    }

    return stakeholder_map.get(company_name, "unknown")
